{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0be34a7c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>sample_1</th>\n",
       "      <th>sample_2</th>\n",
       "      <th>mean_counts</th>\n",
       "      <th>log2_mean</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MYC</td>\n",
       "      <td>300</td>\n",
       "      <td>280</td>\n",
       "      <td>290.0</td>\n",
       "      <td>8.184875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TP53</td>\n",
       "      <td>120</td>\n",
       "      <td>140</td>\n",
       "      <td>130.0</td>\n",
       "      <td>7.033423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>50</td>\n",
       "      <td>45</td>\n",
       "      <td>47.5</td>\n",
       "      <td>5.599913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>11.0</td>\n",
       "      <td>3.584963</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene  sample_1  sample_2  mean_counts  log2_mean\n",
       "2    MYC       300       280        290.0   8.184875\n",
       "0   TP53       120       140        130.0   7.033423\n",
       "3   EGFR        50        45         47.5   5.599913\n",
       "1  BRCA1        10        12         11.0   3.584963"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "# Simulated gene expression counts\n",
    "df = pd.DataFrame({\n",
    "    \"gene\": [\"TP53\", \"BRCA1\", \"MYC\", \"EGFR\"],\n",
    "    \"sample_1\": [120, 10, 300, 50],\n",
    "    \"sample_2\": [140, 12, 280, 45]\n",
    "})\n",
    "\n",
    "# Compute mean expression\n",
    "df[\"mean_counts\"] = df[[\"sample_1\", \"sample_2\"]].mean(axis=1)\n",
    "\n",
    "# Log transform (common in RNA-seq)\n",
    "df[\"log2_mean\"] = np.log2(df[\"mean_counts\"] + 1)\n",
    "\n",
    "df.sort_values(\"mean_counts\", ascending=False)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bio",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
